opencensus/exporters/trace/zipkin/zipkin_exporter.cc

```diff
@@ -0,0 +1,342 @@
+    // Failed to set request timeout.
+    error = curl_easy_strerror(res);
+  } else if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_NOSIGNAL, 1))) {
+    // Failed to disable signals.
+    error = curl_easy_strerror(res);
+  } else {
+    if (proxy_.empty()) {
+      if (CURLE_OK !=
+          (res = curl_easy_setopt(curl, CURLOPT_PROXY, proxy_.c_str()))) {
+        // Failed to set proxy.
+        error = curl_easy_strerror(res);
+      }
+
+      if (http_proxy_tunnel_) {
+        if (CURLE_OK !=
+            (res = curl_easy_setopt(curl, CURLOPT_PROXYTYPE, CURLPROXY_HTTP))) {
+          // Failed to set HTTP proxy type.
+          error = curl_easy_strerror(res);
+        } else if (CURLE_OK !=
+                   (res = curl_easy_setopt(curl, CURLOPT_HTTPPROXYTUNNEL, 1))) {
+          // Failed to set HTTP proxy tunnel.
+          error = curl_easy_strerror(res);
+        }
+      }
+    }
+
+    if (max_redirect_times_) {
+      if (CURLE_OK !=
+          (res = curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1))) {
+        // Failed to enable follow location.
+        error = curl_easy_strerror(res);
+      } else if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_MAXREDIRS,
+                                                     max_redirect_times_))) {
+        // Failed to set max redirect times.
+        error = curl_easy_strerror(res);
+      }
+    }
+
+    // Sending HTTP request to url.
+    if (CURLE_OK != (res = curl_easy_perform(curl))) {
+      // Failed to send http request.
+      error = curl_easy_strerror(res);
+    } else {
+      long status_code = 0;
+      double total_time = 0, uploaded_bytes = 0, upload_speed = 0;
+
+      if (CURLE_OK != (res = curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE,
+                                               &status_code))) {
+        // Failed to get status code.
+        error = curl_easy_strerror(res);
+      } else if (CURLE_OK != (res = curl_easy_getinfo(curl, CURLINFO_TOTAL_TIME,
+                                                      &total_time))) {
+        // Failed to get total time.
+        error = curl_easy_strerror(res);
+      } else if (CURLE_OK !=
+                 (res = curl_easy_getinfo(curl, CURLINFO_SIZE_UPLOAD,
+                                          &uploaded_bytes))) {
+        // Failed to get uploaded bytes.
+        error = curl_easy_strerror(res);
+      } else if (CURLE_OK !=
+                 (res = curl_easy_getinfo(curl, CURLINFO_SPEED_UPLOAD,
+                                          &upload_speed))) {
+        // Failed to get upload speed.
+        error = curl_easy_strerror(res);
+      } else {
+        // LOG(INFO) << "HTTP request finished, status " << status_code
+        //           << ", uploaded " << uploaded_bytes << " bytes in "
+        //           << total_time << " seconds (" << (upload_speed / 1024)
+        //           << " KB/s)";
+      }
+    }
+  }
+
+  // TODO: Log error if one exists.
```

Comment:
Maybe: `if (!error.empty()) std::cerr << "curl error: " << error << "\n";` for now.

Reply:
Done.